entity-framework/core/what-is-new/ef-core-6.0/whatsnew.md

```diff
@@ -2,30 +2,767 @@
+
+EF Core now supports:
+
+* The creation of temporal tables using Migrations
+* Transformation of existing tables into temporal tables, again using Migrations
+* Querying historical data
+* Restoring data from some point in the past
+
+### Configuring a temporal table
+
+The model builder can be used to configure a table as temporal. For example:
+
+<!--
+modelBuilder
+    .Entity<Employee>()
+    .ToTable("Employees", b => b.IsTemporal());
+-->
+[!code-csharp[SimpleConfig](../../../../samples/core/Miscellaneous/NewInEFCore6/TemporalTablesSample.cs?name=SimpleConfig)]
+
+When using EF Core to create the database, the new table will be configured as a temporal table with the SQL Server defaults for the timestamps and history table. For example, consider an `Employee` entity type:
+
+<!--
+    public class Employee
+    {
+        public Guid EmployeeId { get; set; }
+        public string Name { get; set; }
+        public string Position { get; set; }
+        public string Department { get; set; }
+        public string Address { get; set; }
+        public decimal AnnualSalary { get; set; }
+    }
+-->
+[!code-csharp[Employee](../../../../samples/core/Miscellaneous/NewInEFCore6/TemporalTablesSample.cs?name=Employee)]
+
+The temporal table created will look like this:
+
+```sql
+DECLARE @historyTableSchema sysname = SCHEMA_NAME()
+EXEC(N'CREATE TABLE [Employees] (
+    [EmployeeId] uniqueidentifier NOT NULL,
+    [Name] nvarchar(100) NULL,
+    [Position] nvarchar(100) NULL,
+    [Department] nvarchar(100) NULL,
+    [Address] nvarchar(1024) NULL,
+    [AnnualSalary] decimal(10,2) NOT NULL,
+    [PeriodEnd] datetime2 GENERATED ALWAYS AS ROW END NOT NULL,
+    [PeriodStart] datetime2 GENERATED ALWAYS AS ROW START NOT NULL,
+    CONSTRAINT [PK_Employees] PRIMARY KEY ([EmployeeId]),
+    PERIOD FOR SYSTEM_TIME([PeriodStart], [PeriodEnd])
+) WITH (SYSTEM_VERSIONING = ON (HISTORY_TABLE = [' + @historyTableSchema + N'].[EmployeeHistory]))');
+```
+
+Notice that SQL Server creates two hidden `datetime2` columns called `PeriodEnd` and `PeriodStart`. These "period columns" represent the time range during which the data in the row existed. These columns are mapped to [shadow properties](xref:core/modeling/shadow-properties) in the EF Core model, allowing them to be used in queries as shown later.
+
+> [!IMPORTANT]
+> The times in these columns are always UTC time generated by SQL Server. UTC times are used for all operations involving temporal tables, such as in the queries shown below.
+
+Notice also that an associated history table called `EmployeeHistory` is created automatically. The names of the period columns and history table can be changed with additional configuration to the model builder. For example:
+
+<!--
```

Comment:
It's intended to be quite detailed. Most of it will be redundant once we have the true docs, but that's okay because this captures a point-in-time and is tailored for people who want to know all about what's new EF Core 6.0, rather than what can be done with EF Core overall once 6.0 is released.